tracing-attributes/src/lib.rs

```diff
@@ -973,26 +1137,85 @@ fn path_to_string(path: &Path) -> String {
     res
 }
 
-// A visitor struct replacing the "self" and "Self" tokens in user-supplied fields expressions when
-// the function is generated by async-trait.
-struct SelfReplacer {
-    ty: Option<syn::TypePath>,
+// A visitor struct to replace idents and types in some piece
+// of code (e.g. the "self" and "Self" tokens in user-supplied
+// fields expressions when the function is generated by an old
+// version of async-trait).
+struct KnifeReplacer<'a> {
+    types: Vec<(&'a str, TypePath)>,
+    idents: Vec<(Ident, Ident)>,
 }
 
-impl syn::visit_mut::VisitMut for SelfReplacer {
+impl<'a> syn::visit_mut::VisitMut for KnifeReplacer<'a> {
+    #[allow(clippy::cmp_owned)]
     fn visit_ident_mut(&mut self, id: &mut Ident) {
-        if id == "self" {
-            *id = Ident::new("_self", id.span())
+        for (old_ident, new_ident) in &self.idents {
+            // we deliberately compare strings because we want to ignore the spans
+            // If we apply clippy's lint, the behavior changes
```

Comment:
nit: can we maybe move the lint to the specific expression where we want to ignore the lint, like:
```rust
            // we deliberately compare strings because we want to ignore the spans
            // If we apply clippy's lint, the behavior changes
             #[allow(clippy::cmp_owned)]
```
or does that not work?

Reply:
Oh yeah, I remember why I didn't do that: https://github.com/tokio-rs/tracing/pull/1228/checks?check_run_id=2047623466#step:4:274 (slightly old versions of clippy would complain about its position). I will revert to its "bad, but working" position.

Reply:
ah, whoops. that's too bad :(